mmf/utils/distributed.py

```diff
@@ -105,7 +123,11 @@ def gather_tensor(tensor):
         for _ in range(world_size):
             tensor_list.append(torch.zeros_like(tensor))
 
-        dist.all_gather(tensor_list, tensor)
+        if is_xla():
+            tensor_list = xm.all_gather(tensor)
+            tensor_list = tensor_list.view(world_size, *tensor.size())
```

Comment:
why this re-shaping?

Reply:
just `return xm.all_gather(tensor, dim=0)`

Reply:
and take it  at the beginning of `with`. no need to define `tensor_list` if xla.